syntax = "proto3";

package platform;

// 心跳消息
message msg_heartbeat
{
	int64     time = 1;  // time
}

// 链接成功消息
message msg_connected
{
	int32     state = 1;  // state
	int64     time = 2;  // time
}

// 登录消息
message msg_login_request
{
	int32	device		= 1;	// device
	string  username	= 2;	// username
	string  userpsd		= 3;	// userpsd
}

message msg_login_reponse
{
	string  msg			= 1;	// msg
}

// 登出消息
message msg_logout_request
{
	int32	device		= 1;	// device
	int32   uid			= 2;	// uid
}

message msg_logout_reponse
{
	string  msg			= 1;	// msg
}

// 聊天消息
message msg_chat_request
{
	int32		device		= 1;  // device
	int32		from_uid	= 2;  // from_uid
	int32		to_uid		= 3;  // to_uid
	string		msg			= 4;  // msg
}

message msg_chat_reponse
{
	int32		from_uid	= 1;  // from_uid
	int32		to_uid		= 2;  // to_uid
	int32		result		= 3;  // result
}

// 用户信息消息
message msg_userinfo_request
{
	int32     uid = 1;			// uid
}

message msg_userinfo_reponse
{
	int32     uid = 1;			// uid
	int32     age = 2;			// u_id
	int32     sex = 3;			// u_id
	string    name = 4;			// u_id
	string    address = 5;		// str
}

